pdc_client/__init__.py

```diff
@@ -254,21 +251,48 @@ class _BeanBagWrapper(object):
        This class wraps attributes and items of Beanbag to let page_size of
        PDCClient's constructor work.
     """
+
     def __init__(self, client, page_size):
         self.client = client
         self.page_size = page_size
 
     def __call__(self, *args, **kwargs):
         if 'page_size' not in kwargs:
             kwargs['page_size'] = self.page_size
-        return self.client(*args, **kwargs)
+        result = self.client(*args, **kwargs)
+        return result
 
     def __getattr__(self, name):
         return _BeanBagWrapper(self.client.__getattr__(name), self.page_size)
 
     def __getitem__(self, *args, **kwargs):
         return _BeanBagWrapper(self.client.__getitem__(*args, **kwargs), self.page_size)
 
+    def results(self, *args, **kwargs):
+        """
+           Return an iterator with all pages of data when multi pages.
+           Return a list when there is a single page.
+        """
+        def worker():
+            kwargs['page'] = 1
+            try:
+                while True:
+                    response = self.client(*args, **kwargs)
+                    if isinstance(response, list):
+                        yield response
+                    else:
+                        yield response['results']
+                    if isinstance(response, dict) and response['next']:
+                        if response.keys() != ['count', 'previous', 'results', 'next']:
+                            raise Exception("No expected result %s" % response)
```

Comment:
This should be specialized exception (`NoResultsError`?) which would allow user to process the unexpected response (so don't stringify it).

Reply:
Change the logic, and add a new NoResultsError here as there is no builtin type.